data/src/shard_interface.rs

```diff
@@ -272,7 +272,16 @@ impl SessionShardInterface {
             shard_uploads.spawn(
                 async move {
                     debug!("Uploading shard {shard_prefix}/{:?} from staging area to CAS.", &si.shard_hash);
-                    let data: bytes::Bytes = std::fs::read(&si.path)?.into();
+                    let mut data: bytes::Bytes = std::fs::read(&si.path)?.into();
```

Comment:
You can just use `si.shard.metadata` -- the shard_info you load below is already present in `si.shard`.